benchmarks-api/src/main/java/io/scalecube/benchmarks/BenchmarksState.java

```diff
@@ -239,12 +247,18 @@ public final void runForAsync(Function<SELF, Function<Long, Publisher<?>>> func)
 
```

Comment:
As you have said runForAsync doesn't work well with this `parallel() runOn() `  trick . Plz revert .